app/services/record_participant_event.rb

```diff
@@ -1,16 +1,14 @@
 # frozen_string_literal: true
 
 require "initialize_with_config"
+require "json_schema/validate_participant_event"
 
 class RecordParticipantEvent
   include InitializeWithConfig
-  class << self
-    def required_params
-      %i[participant_id lead_provider declaration_type declaration_date raw_event]
-    end
-  end
+  required_config :participant_id, :lead_provider, :declaration_type, :declaration_date, :raw_event
 
   def call
+    return :unprocessable_entity unless schema_validation
```

Comment:
Ok. Good point. It raises to set the exception error, which means that it will only hit this line if successful. I was considering making that schema based validation a service on its own which is called before this one. I flip-flopped on the "validation" and "it's part of the pipeline" a couple of times in my head.